File: 0701_0800/0731/0731.py
from math import floor, ceil
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    d = []
    s, ps = 0, 0
    for i in range(n):
        line = f.readline().split()
        d.append([line[0] == '+', int(line[1]), None])
        s += int(line[1])
    for x in d:
        x[2] = ceil(100 * x[1] / s) if x[0] else floor(100 * x[1] / s)
        ps += x[2]
    for x in d:
        if ps == 100:
            break
        elif ps < 100 and not x[0] and 100 * x[1] % s:
            x[2], ps = x[2] + 1, ps + 1
        elif ps > 100 and x[0] and 100 * x[1] % s:
            x[2], ps = x[2] - 1, ps - 1
    q.write('\n'.join(str(x[2]) for x in d))

File: 0701_0800/0731/0731.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

typedef struct product
{
	int res;
	int quan;
	int good;
} product;

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, s = 0, ps = 0;
	product *d;

	fscanf(f,"%d\n", &n);
	d = (product*)malloc(sizeof(product) * n);
	for (int i = 0; i < n; ++i)
	{
		char c;
		fscanf(f, "%c %d\n", &c, &d[i].quan);
		d[i].good = (c == '+');
		s += d[i].quan;
	}
	for (int i = 0; i < n; ++i)
		ps += (d[i].res = d[i].good ? ceil(100.0 * d[i].quan / s) : floor(100.0 * d[i].quan / s));
	for (int i = 0; i < n && ps != 100; ++i)
	{
		if (ps < 100 && !d[i].good && 100 * d[i].quan % s)
			++d[i].res, ++ps;
		else if (ps > 100 && d[i].good && 100 * d[i].quan % s)
			--d[i].res, --ps;
	}
	for (int i = 0; i < n; ++i)
		fprintf(q, "%d\n", d[i].res);
	return 0;
}
